### PR TITLE
Adds an option to disable auto insert

### PR DESCRIPTION
**Pull request by akhera99**
Roslyn side fix: https://github.com/dotnet/roslyn/pull/74485
To fix: #6349

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -732,6 +732,11 @@
         "title": "Text Editor",
         "order": 1,
         "properties": {
+          "dotnet.autoInsert.enableAutoInsert": {
```
Comment (JoeRobich) [inline package.json:735]: Will auto insert be used for more than doc comments? Perhaps the settings should be `dotnet.autoInsert.insertDocComments` imagining that certain auto inserts might be disabled without disabling all auto inserts.

Comment (JoeRobich) [reply package.json:735]: If we do just want one big switch, then I would suggest `dotnet.autoInsert.enabled`

Comment (Cosifne) [inline package.json:735]: This needs to be covered by test.
https://github.com/dotnet/vscode-csharp/blob/main/test/lsptoolshost/unitTests/configurationMiddleware.test.ts

**Review by dibarbet — COMMENTED** [review]
looks fine - but we should include the server side update in this PR as well

**Review by dibarbet — APPROVED** [review]

**Pull request merged**